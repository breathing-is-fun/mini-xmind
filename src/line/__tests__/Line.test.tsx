import React from 'react';
import { mount } from 'enzyme';
import { SteppedLineTo } from '..';
import 'nino-cli/scripts/setup';

const from = {
  width: 0,
  height: 0,
  left: 0,
  top: 0,
};

const to = {
  width: 0,
  height: 0,
  left: 100,
  top: 0,
};

describe('Line', () => {
  it('Line renders correctly', () => {
    const wrapper = mount(
      <SteppedLineTo from={from} to={to} key="line-test" orientation="v" />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.stepped-line-to').length).toBe(1);
  });

  it('offset works well', () => {
    const wrapper = mount(
      <SteppedLineTo
        from={from}
        to={to}
        key="line-test"
        orientation="v"
        offset={{ x: 100, y: 100 }}
      />,
    );
    expect(wrapper).toMatchSnapshot();

    const x0: string = wrapper.find('SteppedLine').prop('x0');
    expect(parseFloat(x0) + 100).toBe(0);

    const y0: string = wrapper.find('SteppedLine').prop('y0');
    expect(parseFloat(y0) + 100).toBe(0);
  });
});
