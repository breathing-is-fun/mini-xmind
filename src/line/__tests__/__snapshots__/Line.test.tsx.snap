// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line Line renders correctly 1`] = `
<SteppedLineTo
  className=""
  from={
    <div>
      
  
      <div
        style="position: absolute; top: 0px; left: 0px;"
      >
        from
      </div>
      

    </div>
  }
  offset={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  orientation="v"
  style={Object {}}
  to={
    <div>
      

      <div
        style="position: absolute; top: 0px; left: 100px;"
      >
        from
      </div>
      

    </div>
  }
>
  <div
    className="stepped-line-to"
  >
    <SteppedLine
      className=""
      from={
        <div>
          
  
          <div
            style="position: absolute; top: 0px; left: 0px;"
          >
            from
          </div>
          

        </div>
      }
      offset={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      orientation="v"
      style={Object {}}
      to={
        <div>
          

          <div
            style="position: absolute; top: 0px; left: 100px;"
          >
            from
          </div>
          

        </div>
      }
      x0={0}
      x1={0}
      y0={0}
      y1={0}
    >
      <Line
        className=""
        from={
          <div>
            
  
            <div
              style="position: absolute; top: 0px; left: 0px;"
            >
              from
            </div>
            

          </div>
        }
        offset={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        orientation="v"
        style={Object {}}
        to={
          <div>
            

            <div
              style="position: absolute; top: 0px; left: 100px;"
            >
              from
            </div>
            

          </div>
        }
        x0={0}
        x1={0}
        y0={0}
        y1={0}
      >
        <div
          className="line-placeholder"
        >
          <div
            className="line-placeholder-item"
            from={
              <div>
                
  
                <div
                  style="position: absolute; top: 0px; left: 0px;"
                >
                  from
                </div>
                

              </div>
            }
            offset={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            orientation="v"
            style={
              Object {
                "borderTopColor": "#ddd",
                "borderTopStyle": "solid",
                "borderTopWidth": 1,
                "left": "0px",
                "position": "absolute",
                "top": "0px",
                "transform": "rotate(0deg)",
                "transformOrigin": "0 0",
                "width": "0px",
                "zIndex": "1",
              }
            }
            to={
              <div>
                

                <div
                  style="position: absolute; top: 0px; left: 100px;"
                >
                  from
                </div>
                

              </div>
            }
          />
        </div>
      </Line>
    </SteppedLine>
  </div>
</SteppedLineTo>
`;

exports[`Line offset works well 1`] = `
<SteppedLineTo
  className=""
  from={
    <div>
      
  
      <div
        style="position: absolute; top: 0px; left: 0px;"
      >
        from
      </div>
      

    </div>
  }
  offset={
    Object {
      "x": 100,
      "y": 100,
    }
  }
  orientation="v"
  style={Object {}}
  to={
    <div>
      

      <div
        style="position: absolute; top: 0px; left: 100px;"
      >
        from
      </div>
      

    </div>
  }
>
  <div
    className="stepped-line-to"
  >
    <SteppedLine
      className=""
      from={
        <div>
          
  
          <div
            style="position: absolute; top: 0px; left: 0px;"
          >
            from
          </div>
          

        </div>
      }
      offset={
        Object {
          "x": 100,
          "y": 100,
        }
      }
      orientation="v"
      style={Object {}}
      to={
        <div>
          

          <div
            style="position: absolute; top: 0px; left: 100px;"
          >
            from
          </div>
          

        </div>
      }
      x0={-100}
      x1={-100}
      y0={-100}
      y1={-100}
    >
      <Line
        className=""
        from={
          <div>
            
  
            <div
              style="position: absolute; top: 0px; left: 0px;"
            >
              from
            </div>
            

          </div>
        }
        offset={
          Object {
            "x": 100,
            "y": 100,
          }
        }
        orientation="v"
        style={Object {}}
        to={
          <div>
            

            <div
              style="position: absolute; top: 0px; left: 100px;"
            >
              from
            </div>
            

          </div>
        }
        x0={-100}
        x1={-100}
        y0={-100}
        y1={-100}
      >
        <div
          className="line-placeholder"
        >
          <div
            className="line-placeholder-item"
            from={
              <div>
                
  
                <div
                  style="position: absolute; top: 0px; left: 0px;"
                >
                  from
                </div>
                

              </div>
            }
            offset={
              Object {
                "x": 100,
                "y": 100,
              }
            }
            orientation="v"
            style={
              Object {
                "borderTopColor": "#ddd",
                "borderTopStyle": "solid",
                "borderTopWidth": 1,
                "left": "-100px",
                "position": "absolute",
                "top": "-100px",
                "transform": "rotate(0deg)",
                "transformOrigin": "0 0",
                "width": "0px",
                "zIndex": "1",
              }
            }
            to={
              <div>
                

                <div
                  style="position: absolute; top: 0px; left: 100px;"
                >
                  from
                </div>
                

              </div>
            }
          />
        </div>
      </Line>
    </SteppedLine>
  </div>
</SteppedLineTo>
`;
